aea/test_tools/test_cases.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains test case classes based on pytest for AEA end-to-end testing."""
+
+import os
+import shutil
+import subprocess  # nosec
+import sys
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from aea.cli import cli
+from aea.configurations.base import PublicId
+from aea.mail.base import Envelope
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.test_tools.exceptions import AEATestingException
+
+from tests.common.click_testing import CliRunner
```

Comment:
Ok so this is of course an issue. We cannot have this import from tests here. @marcofavorito should we move the `click_testing` file also into the `aea.test_tools`?

In any case, we should make  the `test_tools` dependencies separately installable, like `aea[test]` or something similar.

Reply:
Let's resolve it in a next PR.

Reply:
No I don't want to merge unaddressed comments as we easily forget about them. Please address this.